editor/plugins/bone_map_editor_plugin.cpp

```diff
@@ -0,0 +1,549 @@
+				if (bone_map->is_connected("profile_updated", callable_mp(this, &BoneMapper::recreate_items))) {
+					bone_map->disconnect("profile_updated", callable_mp(this, &BoneMapper::recreate_items));
+				}
+			}
+		}
+	}
+}
+
+BoneMapper::BoneMapper(Skeleton3D *p_skeleton, Ref<BoneMap> &p_bone_map) {
+	skeleton = p_skeleton;
+	bone_map = p_bone_map;
+}
+
+BoneMapper::~BoneMapper() {
+}
+
+void BoneMapEditor::create_editors() {
+	if (!skeleton) {
+		return;
+	}
+	bone_mapper = memnew(BoneMapper(skeleton, bone_map));
+	add_child(bone_mapper);
+}
+
+void BoneMapEditor::fetch_objects() {
+	// Hackey... but it may be the easist way to get a selected object from "ImporterScene".
+	SceneImportSettings *si = SceneImportSettings::get_singleton();
+	if (!si) {
+		return;
+	}
+	Node *selected = si->get_selected_node();
+	if (selected) {
+		Skeleton3D *sk = Object::cast_to<Skeleton3D>(selected);
+		if (!sk) {
+			return;
+		}
+		skeleton = sk;
+	} else {
+		// Editor should not exist.
+		skeleton = nullptr;
+	}
+}
+
+void BoneMapEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			fetch_objects();
+			create_editors();
+		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			remove_child(bone_mapper);
+			bone_mapper->queue_delete();
+		}
+	}
+}
+
+BoneMapEditor::BoneMapEditor(Ref<BoneMap> &p_bone_map) {
+	bone_map = p_bone_map;
+}
+
+BoneMapEditor::~BoneMapEditor() {
+}
+
+bool EditorInspectorPluginBoneMap::can_handle(Object *p_object) {
+	return Object::cast_to<BoneMap>(p_object) != nullptr;
+}
+
+void EditorInspectorPluginBoneMap::parse_begin(Object *p_object) {
+	BoneMap *bm = Object::cast_to<BoneMap>(p_object);
+	if (!bm) {
+		return;
+	}
+	Ref<BoneMap> r(bm);
+	editor = memnew(BoneMapEditor(r));
+	add_custom_control(editor);
+}
+
+BoneMapEditorPlugin::BoneMapEditorPlugin() {
+	// Register properties in editor settings.
+	EDITOR_DEF("editors/bone_mapper/handle_colors/set", Color(0.1, 0.6, 0.25));
+	EDITOR_DEF("editors/bone_mapper/handle_colors/error", Color(0.8, 0.2, 0.2));
+	EDITOR_DEF("editors/bone_mapper/handle_colors/unset", Color(0.3, 0.3, 0.3));
+
+	Ref<EditorInspectorPluginBoneMap> inspector_plugin;
+	inspector_plugin.instantiate();
+	add_inspector_plugin(inspector_plugin);
+
+	Ref<SkeletonPostImportPluginRenamer> post_import_plugin_renamer;
+	post_import_plugin_renamer.instantiate();
+	add_scene_post_import_plugin(post_import_plugin_renamer);
+}
+
+// Buil-in PostImportPlugins
+
+void SkeletonPostImportPluginRenamer::get_internal_import_options(InternalImportCategory p_category, List<ResourceImporter::ImportOption> *r_options) {
+	if (p_category == INTERNAL_IMPORT_CATEGORY_SKELETON_3D_NODE) {
+		r_options->push_back(ResourceImporter::ImportOption(PropertyInfo(Variant::BOOL, "retarget/bone_renamer/rename_bones"), true));
+	}
+}
+
+void SkeletonPostImportPluginRenamer::internal_process(InternalImportCategory p_category, Node *p_base_scene, Node *p_node, Ref<Resource> p_resource, const Dictionary &p_options) {
```

Comment:
I would prefer if this class is its own file. Currently it is mixed with UI code, which I find very confusing.

My personal opinion is that i would prefer it in the editor/import folder.

Reply:
I replaced it into editor/import.